src/discovery/discoveryLogic.ts

```diff
@@ -1,90 +1,247 @@
+export async function loadPreferences(store: LiveStore, user: NamedNode): Promise <NamedNode | undefined > {
+  // console.log('loadPreferences @@ user', user)
   const profile = await loadProfile(store as LiveStore, user)
   const preferencesFile = store.any(user, ns.space('preferencesFile'), undefined, profile)
+  // console.log('loadPreferences @@ pref file', preferencesFile)
   if (!preferencesFile) {
-    // throw new Error(`USer ${user} has no pointer in profile to preferences file.`)
+    const message = `User ${user} has no pointer in profile to preferences file.`
+    console.warn(message)
+    // throw new Error()
     return undefined
   }
   try {
-    store.fetcher.load(preferencesFile as NamedNode)
+    await store.fetcher.load(preferencesFile as NamedNode)
   } catch (err) { // Mabeb a permission propblem or origin problem
     return undefined
     // throw new Error(`Unable to load preferences file ${preferencesFile} of user <${user}>: ${err}`)
   }
   return preferencesFile as NamedNode
 }
 
-export async function loadTypeIndexesFor(store: LiveStore, user:NamedNode): Promise<Array<TypeIndex>> {
+export async function loadTypeIndexesFor(store: LiveStore, user:NamedNode): Promise<Array<TypeIndexScope>> {
+  // console.log('@@ loadTypeIndexesFor user', user)
   if (!user) throw new Error(`loadTypeIndexesFor: No user given`)
   const profile = await loadProfile(store, user)
   const publicTypeIndex = store.any(user, ns.solid('publicTypeIndex'), undefined, profile)
-  if (publicTypeIndex) {
-    try {
-      await store.fetcher.load(publicTypeIndex as NamedNode)
-    } catch {
-      // never mind
-    }
+  // console.log('@@ loadTypeIndexesFor publicTypeIndex', publicTypeIndex)
+
+  const  publicScopes = publicTypeIndex ? [ { label: 'public', index: publicTypeIndex as NamedNode, agent: user } ] : []
+
+  let preferencesFile
+  try {
+    preferencesFile = await loadPreferences(store, user)
+  } catch (err) {
+    preferencesFile = null
   }
-  const  pub = publicTypeIndex ? [ { label: 'public', index: publicTypeIndex as NamedNode, agent: user } ] : []
 
-  const preferencesFile = await loadPreferences(store, user)
+  let priv
   if (preferencesFile) { // watch out - can be in either as spec was not clear
     const privateTypeIndexes = store.each(user, ns.solid('privateTypeIndex'), undefined, preferencesFile as NamedNode)
       .concat(store.each(user, ns.solid('privateTypeIndex'), undefined, profile))
-    const priv = privateTypeIndexes.length > 0 ? [ { label: 'priSo @@@@@vate', index: privateTypeIndexes[0] as NamedNode, agent: user } ] : []
-    return pub.concat(priv)
+     // console.log('@@ loadTypeIndexesFor privateTypeIndexes', privateTypeIndexes)
+
+    priv = privateTypeIndexes.length > 0 ? [ { label: 'private', index: privateTypeIndexes[0] as NamedNode, agent: user } ] : []
+  } else {
+    priv = []
+  }
+  const scopes =  publicScopes.concat(priv)
+  if (scopes.length === 0) return scopes
```

Comment:
If there is no `publicTypeIndex` we do not continue to private? Should it be a precondition?